lib/core_extensions.rb

```diff
@@ -39,8 +39,14 @@ def initialize(*attribute)
 
     def before_destroy(record)
       klasses.each do |klass|
-        record.send(klass.to_sym).each do |what|
-          what = what.to_label unless what.is_a? String
+        record.association(klass.to_sym).association_scope.each do |what|
```

Comment:
Would doing `assocation_scope.unscoped` work?  (It ought to have no effect on Rails 3, but probably would on 4.)